File: solutions/0645-set-mismatch/solution.py
class Solution(object):
    def findErrorNums(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        ls=[]
        n=sum(list(set(nums)))
        ls.append(sum(nums)-n)
        x=(len(nums)*(len(nums)+1))/2
        ls.append(x-n)
        return ls

File: solutions/0645-set-mismatch/solution.java
class Solution {
    public int[] findErrorNums(int[] nums) {
        int arr[]=new int[2];
        int a=0,flag=1,n=nums.length;
        for(int i=0;i<n;i++)
        {
            int val=Math.abs(nums[i]);
            a+=val;
            if(nums[val-1]<0 && flag-- >0)
                arr[0]=val;
            else
                nums[val-1]*=-1;
        }
        int org=(n*(n+1))/2;
        a-=arr[0];
        arr[1]=org-a;
        return arr;
    }
}
